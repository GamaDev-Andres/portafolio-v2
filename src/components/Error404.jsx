import { Link } from 'react-router-dom'
import NavProject from './project/NavProject'

const Error404 = () => {
  return (
    <>
      <NavProject />
      <div className=" flex items-center justify-center flex-col gap-4 p-2">
        <div className="p-4 border border-blue-500 shadow-md shadow-blue-300">
          <h2 className="text-blue-400 font-bold text-2xl">¡Error 404!</h2>
          <p>La página a la que intento acceder no existe.</p>
          <Link
            to="/"
            className="block py-2 px-4 rounded-md text-center ring-1 mt-4"
          >
            Página principal
          </Link>
        </div>
      </div>
    </>
  )
}

export default Error404
