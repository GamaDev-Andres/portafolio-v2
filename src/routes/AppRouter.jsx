import { useState } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from '../components/home/Home';
import Project from '../components/project/Project';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/project/:id" element={<Project />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;
