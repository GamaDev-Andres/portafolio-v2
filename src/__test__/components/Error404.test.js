/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { HelmetProvider } from 'react-helmet-async'
import { MemoryRouter } from 'react-router-dom'
import Error404 from '../../components/Error404'
import AppRouter from '../../routes/AppRouter'

describe('probando <Error404/>', () => {
  test('debe montarse correctamente', () => {
    const wrapper = Enzyme.mount(
      <MemoryRouter>
        <Error404 />
      </MemoryRouter>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('h2').text().trim()).toBe('Â¡Error 404!')
  })
  test('debe montarse en un ruta no existente', () => {
    const wrapper = Enzyme.mount(
      <MemoryRouter initialEntries={['/edeefeefefef']}>
        <HelmetProvider>
          <AppRouter />
        </HelmetProvider>
      </MemoryRouter>
    )
    expect(wrapper.find('Error404').exists()).toBe(true)
  })
})
