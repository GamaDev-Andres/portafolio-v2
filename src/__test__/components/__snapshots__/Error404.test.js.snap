// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <Error404/> debe montarse correctamente 1`] = `
Array [
  <nav
    className="fixed w-full  sm:top-0 sm:left-0 z-10"
  >
    <div
      className="flex relative justify-center max-w-[1256px] mx-auto bg-base dark:bg-baseDark items-center px-4 py-2"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1
          className="text-letra font-black text-4xl px-4 text-center dark:text-letraDark"
        >
          GamaDev
        </h1>
      </a>
      <a
        className="absolute top-4 right-4"
        href="/"
        onClick={[Function]}
      >
        <i
          className="sm:mr-2 fa-solid fa-house"
        />
      </a>
    </div>
  </nav>,
  <div
    className=" flex items-center justify-center flex-col gap-4 p-2"
  >
    <div
      className="p-4 border border-blue-500 shadow-md shadow-blue-300"
    >
      <h2
        className="text-blue-400 font-bold text-2xl"
      >
        ¡Error 404!
      </h2>
      <p>
        La página a la que intento acceder no existe.
      </p>
      <a
        className="block py-2 px-4 rounded-md text-center ring-1 mt-4"
        href="/"
        onClick={[Function]}
      >
        Página principal
      </a>
    </div>
  </div>,
]
`;
